### PR TITLE
Add testCase to handle stream only with new events

### PR DESCRIPTION
**Pull request by rabbl**
I added a testcase to increase the linecoverage in the eventStoreProjection.
If there are no new events, a NoNewerEventsFound-Exeption will be thrown and catched.
So the loop continues checking the next stream.
Please note, there is a possible case, that the projection cannot be stopped.

### CODE REVIEW

File: src/Projection/InMemoryEventStoreProjection.php [hunk 1]
```diff
@@ -328,8 +328,7 @@ public function run(bool $keepRunning = true): void
             foreach ($this->streamPositions as $streamName => $position) {
                 try {
                     $streamEvents = $this->eventStore->load(new StreamName($streamName), $position + 1);
-                } catch (Exception\StreamNotFound $e) {
-                    // no newer events found
+                } catch (Exception\NoNewerEventsFound $e) {
```
Comment (rabbl) [inline src/Projection/InMemoryEventStoreProjection.php]: The isStoped-conditional could go outside of the foreach-loop.

File: src/InMemoryEventStore.php [hunk 2]
```diff
@@ -151,6 +152,10 @@ public function load(
             }
         }
 
+        if (count($streamEvents) == 0) {
+            throw NoNewerEventsFound::with($streamName);
```
Comment (prolic) [inline src/InMemoryEventStore.php]: why is this needed? i am okay with returning a empty iterator.

**Comment by rabbl** [comment]
The InMemoryEventStoreTest is failing now, because the the throws exception is unexpected.
I can fix this, but I'm not very convinced that I did it in the right way.

File: src/InMemoryEventStore.php (hunk 2 above)
Comment (rabbl) [inline src/InMemoryEventStore.php]: I have tried to write a test to increase the linecoverage here:
https://coveralls.io/builds/10114395/source?filename=src%2FProjection%2FInMemoryEventStoreReadModelProjection.php#L298
As I could see, there was no test possible to throw this exception.
So my idea war to add an eventsNotFound exception instead and catch it.

Comment (rabbl) [inline src/InMemoryEventStore.php]: Whats your idea in this case?

Comment (prolic) [inline src/InMemoryEventStore.php]: @rabbl this exception catching can be removed completely. No StreamNotFound exception is thrown any more, when no newer events are found - an empty iterator is returned instead, so the catching can be removed. wanna PR this?

Comment (rabbl) [inline src/InMemoryEventStore.php]: Done! 
https://github.com/prooph/event-store/pull/260
